OCP-1008: migrated to node 22

azavada opened the pull request:
Checklist:

- [ ] internal documentation is up-to-date
- [ ] external documentation is up-to-date

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ZaiusInc/app-sdk/144)
<!-- Reviewable:end -->

pawelziebaopti commented on .github/node.env:
```diff
@@ -0,0 +1 @@
+NODE_VERSION=22.x
```
this is not needed in this repo I guess as the script in node-sdk uses major version in package.json to determinate node version.

pawelziebaopti reviewed (commented):
are all dependencies upgraded? if not, is there a documentation on which are not any why?

azavada commented:
> are all dependencies upgraded? if not, is there a documentation on which are not any why?

Not all

- `node-fetch`: Updated to the latest version that supports CommonJS import syntax. The newest version uses ESM syntax, which has only experimental support in jest, so we opted for the CommonJS version for compatibility.
- `remark`: Not updated as it is already at the latest version that uses CommonJS syntax, ensuring compatibility with our current setup.
- `remark-validate-links`: Updated to the latest version that remains compatible with our existing setup and other dependencies.

azavada commented on tsconfig.json:
```diff
@@ -6,9 +6,9 @@
     "outDir": "./dist",
     "declaration": true,
     "moduleResolution": "node",
```
Well, I'm not sure about incompatibility. This option tells TS how to import modules. It is good practice to use the latest recommended option. However, the new Node JS version works fine with this. Anyway, I tried to use the latest option, and the "compiler" gave me many errors, which I struggled to fix. However, after a fresh look today, I could use the same config settings as in the node-sdk and fix all the code errors.

pawelziebaopti commented on src/app/types/AppManifest.ts:
```diff
@@ -51,7 +51,7 @@ export type AppCategory = 'Commerce Platform' | 'Point of Sale' | 'Lead Capture'
 | 'Subscriptions';
 
-export type AppRuntime = 'node12' | 'node18' | 'node18_rt';
+export type AppRuntime = 'node12' | 'node18' | 'node18_rt' | 'node22' | 'node22_rt';
```
are you planning to migrate 'rt' runtime as well? not sure if that's needed as this type of runtime doesn't have wide adoption. not sure if we need to migrate it.

azavada replied:
Weill, it is worth a try. However, I don't know how deep the rabbit hole is...
